lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,278 @@
+<template>
+	<div
+		class="ds-datePicker"
+		:class="{ '-ds-box': triggerType === DATE_PICKER_TRIGGER_TYPES.BOX }"
+	>
+		<template v-if="triggerType === DATE_PICKER_TRIGGER_TYPES.TILE">
+			<span
+				class="ds-datePicker__label"
+				:class="{
+					'-ds-isUppercase': isLabelUppercase,
+					'-ds-disabled': state === DATE_PICKER_STATES.DISABLED,
+				}"
+				>{{ label }}</span
+			>
+			<ds-tile
+				:text="text"
+				:interactive="isInteractive"
+				:color="color"
+				:state="state"
+				:icon-right="icon"
+				:is-icon-right-hidden-on-mobile="isIconHiddenOnMobile"
+				:eyebrow-text="eyebrowText"
+				@click="toggle"
+			/>
+		</template>
+		<date-picker-box
+			v-else
+			:is-interactive="isInteractive"
+			:placeholder="placeholder"
+			:date="date"
+			:icon="icon"
+			:is-icon-hidden-on-mobile="isIconHiddenOnMobile"
+			:state="state"
+			:color="color"
+			:eyebrow-text="eyebrowText"
+			:is-open="isOpen"
+			@click="toggle"
+		/>
+
+		<span class="ds-datePicker__errorMessage"> {{ errorMessage }} </span>
+		<input ref="flatpickrInput" class="ds-datePicker__hiddenInput" />
+	</div>
+</template>
+
+<style lang="scss">
```

Comment:
```suggestion
<style lang="scss" scoped>
```

Reply:
It cannot be scoped because flatpickr theme styles won't work

Reply:
then let's separate them

Reply:
What the separation gives us in that matter?

Reply:
is that we have a separation of styles and they do not leak to the entire application
```
<style lang="scss">
@import 'flatpickr/dist/flatpickr';
</style>

<style lang="scss" scoped>
@import '../../../../styles/settings/typography/tokens';
@import '../../../../styles/settings/colors/tokens';
@import '../../../../styles/settings/spacings';
```